src/main/java/net/dries007/tfc/common/blocks/plant/BodyPlantBlock.java

```diff
@@ -0,0 +1,49 @@
+package net.dries007.tfc.common.blocks.plant;
+
+import java.util.function.Supplier;
+
+import net.minecraft.block.*;
+import net.minecraft.tags.BlockTags;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.shapes.VoxelShape;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.IWorldReader;
+
+public class BodyPlantBlock extends AbstractBodyPlantBlock
+{
+    private final Supplier<? extends Block> headBlock;
+
+    public BodyPlantBlock(AbstractBlock.Properties properties, Supplier<? extends Block> headBlock, VoxelShape shape, Direction direction)
+    {
+        super(properties, direction, shape, false);
+        this.headBlock = headBlock;
+    }
+
+    protected AbstractTopPlantBlock getHeadBlock()
+    {
+        return (AbstractTopPlantBlock) headBlock.get();
+    }
+
+    @Override
+    public boolean isValidBonemealTarget(IBlockReader worldIn, BlockPos pos, BlockState state, boolean isClient)
+    {
+        return false;
+    }
```

Comment:
There's ~4 methods for bonemeal, as per `IGrowable`. For consistency if we override one in the negative, we should no-op or return false on all. (Despite the aforementioned comment about indev bonemeal handling.